### PR TITLE
Task/33 validations

### PR DESCRIPTION
**Pull request by sanzog03**

### CODE REVIEW

**Comment by SahSantoshh** [comment]
This doesn't seem a proper message to be sent to front-end

    {
    "status": "error",
    "statusCode": 422,
    "message": "Invalid Input",
    "errors": [
        {
            "data.attributes.name": "WHERE parameter \"name\" has invalid \"undefined\" value"
        }
      ]
     }

**Comment by SahSantoshh** [comment]
No error message for user post

    {
    "status": "error",
    "statusCode": 400,
    "message": "SequelizeUniqueConstraintError",
    "errors": ""
    }

**Comment by SahSantoshh** [comment]
And this as well

    {
    "status": "error",
    "statusCode": 422,
    "message": "Invalid Input",
    "errors": "notNull Violation: AutomationScheduler.AlgorithmStartTime_start cannot be null"
    }

**Comment by sanzog03** [comment]
message is provided in general format easily usable and popable by frontend. error attribute contains the detailed error message

**Comment by sanzog03** [comment]
'No error message for user post` yo chai kun route ko lagi ho exactly??

**Comment by sanzog03** [comment]
now error is formatted as `{
    "status": "error",
    "statusCode": 422,
    "message": "Invalid Input",
    "errors": [
        {
            "message": "Role already exists"
        }
    ]
}`

**Comment by sanzog03** [comment]
see `message` instead of `attribute.data.user`

**Comment by sanzog03** [comment]
also added in validator when there is no data during post/put routes for important keys in body

**Comment by sanzog03** [comment]
i think i've done the things that we discussed eariler. please have a check and go through it and add any if you feel so.

File: validator/role.validator.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const { body, validationResult } = require("express-validator");
+const { ErrorHandler } = require("../helpers/error");
+const { models } = require("../models").dbConnectionOne;
+
+const validationRules = () => {
+  return [
+    body("data.attributes.name").custom((name) => {
+      if (!name) {
+        return Promise.reject("Name field is empty");
```
Comment (sanzog03) [inline validator/role.validator.js:9]: empty name handled

File: validator/user.role.validator.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      if (!roleId) {
+        return Promise.reject("RoleId is empty");
+      }
+      return models.role.findOne({ where: { id: roleId } }).then((role) => {
+        if (!role) {
+          return Promise.reject("Role does not exist");
+        }
+      });
+    }),
+    param("id").custom((id) => {
```
Comment (sanzog03) [inline validator/user.role.validator.js:17]: empty id in params handled

**Comment by SahSantoshh** [comment]
This validation hits db too many times it need to be show db errors

**Pull request closed without merging**